File: fixtures/csharp/type_inference/control_positions/control_positions.cs
namespace PositionApp
{
    public class ControlPositions
    {
        public static string HandleWhile(object value)
        {
            var current = value;
            while (current is string) {
                current = ((string)current).ToUpper();
            }
            return current.ToString();
        }

        public static string HandleElseIf(object value)
        {
            if (value is string) {
                return (string)value;
            } else if (value is int) {
                return "number";
            }
            return "other";
        }
    }
}

File: fixtures/java/type_inference/control_positions/control_positions.java
public class ControlPositions {

    public static String handleWhile(Object value) {
        Object current = value;
        while (current instanceof String) {
            current = ((String) current).toUpperCase();
        }
        return current.toString();
    }

    public static String handleElseIf(Object value) {
        if (value instanceof String) {
            return (String) value;
        } else if (value instanceof Integer) {
            return "number: " + value;
        }
        return "other";
    }

    public static void main(String[] args) {
        System.out.println(handleWhile("hi"));
        System.out.println(handleElseIf(42));
    }
}
